doc/user-guide.adoc

```diff
@@ -29,6 +29,72 @@ The `applicationImage` accepts different references to Docker image:
 * a tag: `quay.io/wildfly-quickstarts/wildfly-operator-quickstart:18.0`
 * a digest: `quay.io/wildfly-quickstarts/wildfly-operator-quickstart@sha256:0af38bc38be93116b6a1d86a9c78bd14cd527121970899d719baf78e5dc7bfd2`
 
+[[source-repository]]
+## Specify the Application Source
+
+The WildFly Operator can also take as input the source code of the application and use https://github.com/openshift/source-to-image[OpenShift Source-to-Image (S2I)] to build the Java applications and create a Docker image from it.
+
+[IMPORTANT]
+====
+This feature is available on OpenShift only.
+
+It relies on OpenShift-specific resources such as `BuildConfig` and `ImageStreams` that are not available on a plain Kubernetes cluster.
+====
+
+[source,yaml]
+.Example of application source configuration
+----
+apiVersion: wildfly.org/v1alpha1
+kind: WildFlyServer
+metadata:
+  name: quickstart
+spec:
+  applicationSource:
+    # The application will be built from the Web application at
+    # https://github.com/wildfly/quickstart/tree/19.0.0.Final/helloworld-rs
+    sourceRepository:
+      url: "https://github.com/wildfly/quickstart.git"
+      ref: "19.0.0.Final"
+    source2Image:
+      builderImage: "wildfly-centos7:19.0"
+      runtimeImage: "wildfly-runtime-centos7:19.0"
+      env:
+      # cloud-server provision a WildFly server that can runs JAX-RS application
+      - name: GALLEON_PROVISION_LAYERS
+        value: "cloud-server"
+      - name: MAVEN_ARGS
+        value: "install -pl org.wildfly.quickstarts:helloworld-rs -am "
+      - name: ARTIFACT_DIR
+        value: "helloworld-rs/target"
+  replicas: 1
+
+----
+
+The application will be built from the project at https://github.com/wildfly/quickstart/tree/19.0.0.Final/helloworld-rs (`ref` corresponds to the branch or tag to checkout). The Git repository at https://github.com/wildfly/quickstart contains many Java application but we only want to build and run one of them from the module
+`helloworld-rs`. We provide the `MAVEN_ARGS`and `ARTIFACT_DIR` environment variables to S2I to build only this Web application.
+
+[NOTE]
+====
+To build the application image, the WildFly S2I images must be installed in the OpenShift cluster before the WildFly Operator deploys the application.
+As an example, to install the WildFly 19 S2I images in the `openshift` cluster, you can run the commands:
+
+[source,shell]
+----
+$ oc import-image wildfly/wildfly-centos7:19.0 --from=quay.io/wildfly/wildfly-centos7:19.0 -n openshift --confirm
+$ oc import-image wildfly/wildfly-runtime-centos7:19.0 --from=quay.io/wildfly/wildfly-runtime-centos7:19.0 -n openshift --confirm
+----
+====
+
+Once the application image is build and deployed, the Operator will install it on the OpenShift cluster and create a Route for it.
```

Comment:
Will there be also some part of documentation for Phase2 features, eg. example of how to trigger update of images by importing new tag to imagestream or github webhook?

Reply:
I've added such doc in https://github.com/jmesnil/wildfly-operator/blob/27_imageStream_runtime/doc/user-guide.adoc#application-source